projects/table-builder/src/lib/functions/rxjs-operators.ts

```diff
@@ -31,3 +31,45 @@ export const combineArrays = <T>(sources: Observable<T[]>[]): Observable<T[]> =>
     map( res => _.flatten(res) )
   );
 };
+
+export function switchOff( switchSource: Observable<boolean>, defaultState: boolean = true) {
+  return <T>(source: Observable<T>) : Observable<T> => {
+    return new Observable(subsciber => {
+      let isOn = defaultState;
+      const subscription = new Subscription();
+      subscription.add( switchSource.subscribe( on => isOn = on ));
```

Comment:
I would assume that there is no guarantee as to the timing. The `next()` on the `switchSource` maybe be delayed (with a `scheduler` or something

Reply:
good point, but I'm not really sure what else can be done.
Like is there a way to force them both onto the same scheduler? 
Any suggestions?